specification.md

```diff
@@ -2677,10 +2715,10 @@ Each iteration of the ForEach state should be executed in parallel.
 You can use the `max` property to set the upper bound on how many iterations may run in parallel. The default
 of the `max` property is zero, which places no limit on number of parallel executions.
 
-The `inputCollection` property is a JsonPath expression which selects an array in the states data. All iterations 
-are done against data elements of this array. This array must exist.
+The `inputCollection` property is a workflow expression which selects an array in the states data. All iterations 
+are performed against data elements of this array. This array must exist.
```

Comment:
Should the workflow fail or throw an exception if the array does not exist?  Or can we say the runtime should assume an empty array if the inputCollection would return a non-array value?

Reply:
if inputcollection is empty or does not exist there is nothing to loop over. i think this should be a runtime error and should be able to be handled in the onErrors def. will update to add that part